metomi/rosie/ws_client_auth.py

```diff
@@ -48,11 +48,11 @@
 import sys
```

Comment:
Do we need to worry about PyGTK and PyGObject dependencies? At the moment both are not listed, and not even optional (`extras_requires`)

Reply:
Uh Oh. This may require some thought. This bit of code is unplesantly delicate (c.f. [this nasty PR](https://github.com/metomi/rose/pull/2254)). We definately need some of it to provide the keyring support, but we should probably be ditching some of it where it is used to support the GUI. This may actually make this nasty mess much nicer.

Reply:
Auth caching will have to be re-worked for the new GUI :(

Reply:
I have now removed all these imports. No doubt this will need to be replaced.